File: backend/app/schemas/feedback.py
from pydantic import BaseModel, Field
from typing import Optional

class CreateFeedbackRequest(BaseModel):
    order_id: str
    rating: int = Field(..., ge=1, le=5)
    comment: Optional[str] = None

class FeedbackResponse(BaseModel):
    id: str
    user_id: str
    order_id: str
    rating: int
    comment: Optional[str] = None
    created_at: Optional[str] = None
    customer_name: Optional[str] = None
    order_date: Optional[str] = None

File: frontend/types/feedback.ts
export interface Feedback {
  id: string;
  user_id: string;
  order_id: string;
  rating: number; // 1 - 5 stars
  comment?: string;
  created_at: string;
  customer_name?: string;
  order_date?: string;
}

export interface CreateFeedbackInput {
  order_id: string;
  rating: number;
  comment?: string;
}
